ocfweb/docs/docs/services/hpc/slurm.md

```diff
@@ -0,0 +1,145 @@
+
+Slurm is only accessible while SSHed into `hpcctl.ocf.berkeley.edu`.
+
+Submitting a job to Slurm can be done in one of two ways: through `srun`, and
+through `sbatch`.
+
+When using `srun`, options are supplied using command-line flags, and the job
+is attached to the terminal you run `srun` from. If there is a queue to run
+jobs, the terminal will wait until your job starts running, and if the terminal
+closes, the job will be cancelled.
+
+By contrast, to submit a job using `sbatch`, you must first create a batch file
+that includes options for your job, and the commands that your job will run.
+The batch file is then submitted, and after the job runs, log files with the
+job's output to `stdout` are put into your home directory.
+
+#### When to use `srun` versus `sbatch`
+
+If your job requires interactivity or inputs from the terminal, or you need a
+terminal to test or experiment, use `srun`. Otherwise, use `sbatch`, as you
+don't have to keep your terminal open until the job runs.
+Alternatively, you could use `screen` or `tmux` with `srun` to keep your
+job alive, even if you disconnect from your terminal.
+
+### Resource options
+
+*Some terminology:* Slurm refers to a process as a "task". Even if a single
+process is using multiple threads/CPUs, it still counts as one task.
+
+**By default, without any flags, a job you submit will be allocated one CPU,
+100 MB of RAM, and no GPUs, and will run for at maximum 2 days. In order to
+allocate more resources and time to your job, you must set one or more of these
+flags:**
+
+* `-n` / `--ntasks`:
+    - The number of tasks/processes to allocate. Default is 1.
+* `-c` / `--cpus-per-task`:
+    - The number of CPUs to allocate per task. Default is 1.
+* `--mem`:
+    - The total amount of RAM to allocate. By default, the number supplied is
+assumed to megabytes. However, the prefixes `K`, `M`, `G`, and `T` can be
+appended to the number instead. For example, to allocate 5 gigabytes of ram,
+use `--mem=5G`. Default is 100 megabytes.
+* `--gres` **(Optional)**:
+    - Allocates some GPUs to your job. The format is `--gres=gpu:[optional
+type]:[number to allocate]`. For example, to allocate 2 GPUs of any type, you
+would include `--gres=gpu:2`. To allocate two Nvidia 1080Ti GPUs (our only type
+right now), you would include `--gres=gpu:nv1080:2`. Default is no GPUs.
+* `--t` / `--time` **(Optional)**:
+    - The maximum amount of time your job can take before Slurm forcefully
+kills it. Acceptable time formats include "minutes", "minutes:seconds",
+"hours:minutes:seconds", "days-hours", "days-hours:minutes" and
+"days-hours:minutes:seconds". You cannot set the time limit greater than the
+default, which is 2 days.
+
+### Using `srun`
+
+On `hpcctl.ocf.berkeley.edu`:
+
+```
+srun [command-line flags] [command to run]
+```
+
+For example, to run a job that uses 4 CPUs, 8 GB of RAM, and 1 GPU:
+
+```
+bzh@hpcctl:~$ srun --ntasks=1 --cpus-per-task=4 --mem=8G --gres=gpu:1 echo "Hello world!"
```

Comment:
When users are actually logged in they'll see `username@segfault`. Wonder if we should comment on this...

Reply:
I feel that HPC users are probably technically competent enough to not get confused by it, if they do notice.